### DISCUSSION TITLE
Can we enable imx477 csi camera by default?

### DISCUSSION THREAD

**Discussion by quyentruongngoc** [OE4T/meta-tegra#894] [Q&A] [answered]
My environment:
Board : Jetson Nano ( Developer Kit )
Branch : dunfell (L4T R32.6.1/JetPack 4.6)
Image : core-image-sato (or core-image-minimal)
Camera: https://www.waveshare.com/imx477-12.3mp-camera.htm

Seem imx477 camera is not set by default, I have connected imx477 camera to Jetson Nano but it cannot detected any camera.
I check my dmes log:
`root@qttech:~# dmesg | grep imx`
`[    3.624930] imx219 7-0010: tegracam sensor driver:imx219_v2.0.6`
`[    3.648280] imx219 7-0010: imx219_board_setup: error during i2c read probe (-121)`
`[    3.648346] imx219 7-0010: board setup failed`
`[    3.648395] imx219: probe of 7-0010 failed with error -121`
`[    3.648815] imx219 8-0010: tegracam sensor driver:imx219_v2.0.6`
`[    3.672150] imx219 8-0010: imx219_board_setup: error during i2c read probe (-121)`
`[    3.672211] imx219 8-0010: board setup failed`
`[    3.672258] imx219: probe of 8-0010 failed with error -121`

I can enable imx477 with OS support  from NVIDIA  (used jetson-io ) here: https://developer.nvidia.com/embedded/downloads
I try to copy /opt/nvidia folder to my board (which running core-image-sato image) but It not work (I also added `dtc, dpkg` to `local.conf`). 

Do we have any package to enable imx477 csi camera by default, or package to install jetson-io on yocto build?

Many thanks,

**Comment by madisongh** [comment, marked as answer]
The jetson-io tools are really only usable with NVIDIA's Ubuntu distro.  There isn't really an equivalent mechanism here for making these kinds of configuration changes at runtime. The usual approach with Yocto-style builds is to modify the device tree directly to configure in the hardware you're using.

It would be useful to make use of the device tree overlays on the development kits, though, so I spent a bit of time today looking at how we could make that happen - so you could at least configure at build time which overlays you want to use, and generate an image that would apply them automatically during boot.  I did get it working, but it will take some time to make it more usable.

**Reply by quyentruongngoc** [reply]
Thanks @madisongh for your feedback, I'm still using NVIDIA's Ubuntu distro for developing my software so it fine to me now. I will look into the device tree according to your suggestion.

Thanks for spend your time on my question @madisongh,

**Comment by madisongh** [comment]
I have updated the kernel recipe so you can at least apply the NVIDIA-supplied device tree overlays at build time.
There is a description on [this wiki page](https://github.com/OE4T/meta-tegra/wiki/Using-device-tree-overlays).

**Reply by quyentruongngoc** [reply]
@madisongh , Thank you very much !!!!!
